tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -3867,6 +3953,7 @@ void CGMSHLSLRuntime::FinishCodeGen() {
     for (auto &it : entryFunctionMap) {
       CloneShaderEntry(it.second, it.getKey(), *m_pHLModule);
     }
+    ReplaceConstStaticGlobals(staticConstGlobalInitListMap, staticConstGlobalCtorMap);
```

Comment:
Why restrict this optimization to libraries? Seems like it would be generally useful.